### ISSUE TITLE
Fix request: false positives for AvoidInfiniteRecursion

### ISSUE THREAD

**Issue by stokpop** [jborgers/PMD-jPinpoint-rules#484]
These contructs give false positives:

```java
@Service
@RequiredArgsConstructor
public class RepoClient {

    private final DetailRepository detailRepository;

    public EventDocument save(EventDocument documentToSave) {
        return detailRepository.save(documentToSave);
    }
}
```

and this one:

```java
@Builder
@Getter
@EqualsAndHashCode
@ToString
public final class IssueRecursion implements Comparable<IssueRecursion>{

  private final String name;

  @Override
  public int compareTo(IssueRecursion issue) {
    return name.compareTo(issue.getName());
  }
}
```

and this one:

```java
public class Listener extends ListenerSupport {

    @Override
    public boolean open(Context context) {
        return super.open(context);
    }
}
```

and this one: 

```java
public class Deserializer {

    public Map deserialize(JsonParser p, Context c) {
        ....
        return target;
    }

    public Map deserialize(JsonParser p) {
        Context c = new Context();
        return this.deserialize(p, c);
    }
}
```

**Comment by jborgers** [comment]
additional false positive:
```
class Issue484_onMethodCall {

    public String foo() {
        return other().foo();
    }
}
```

**Comment by jborgers** [comment]
another one: 
```
class Issue484_onThisAttributeCall {
    Other other;
    public String foo() {
        return this.other.foo();
    }
}
```
